dss/api/bundles/checkout.py

```diff
@@ -0,0 +1,65 @@
+import json
+
+import boto3
+import os
+import requests
+from flask import jsonify
+from cloud_blobstore import BlobNotFoundError
+from dss import Config
+
+from dss.util.bundles import get_bundle
+from dss.util.checkout import get_execution_id
+from ... import dss_handler, get_logger
+
+
+@dss_handler
+def post(uuid: str, json_request_body: dict, replica: str, version: str = None):
+
+    sfn = boto3.client('stepfunctions')
+    sts_client = boto3.client("sts")
+
+    email = json_request_body['email']
+    dss_bucket = Config.get_s3_bucket()
+
+    if replica is None:
+        replica = "aws"
+
+    try:
+        get_bundle(uuid, replica, version)
+
+        sfn_input = {"dss_bucket": dss_bucket, "bundle": uuid, "version": version, "email": email, "replica": replica}
+        if "destination" in json_request_body:
+            sfn_input["bucket"] = json_request_body["destination"]
+        execution_input = json.dumps(sfn_input)
+
+        region = boto3.Session().region_name
+        accountid = sts_client.get_caller_identity()['Account']
+        stage = os.environ["DSS_DEPLOYMENT_STAGE"]
+        state_machine_arn = f"arn:aws:states:{region}:{accountid}:stateMachine:dss-checkout-sfn-{stage}"
+        response = sfn.start_execution(
+            stateMachineArn=state_machine_arn,
+            name=get_execution_id(),
+            input=execution_input
+        )
+        return jsonify(dict(checkout_job_id=response["executionArn"])), requests.codes.ok
+    except Exception as e:
```

Comment:
I don't think a generic catch-all Exception should result in a not_found error code.

Reply:
fixed